torchvision/models/regnet.py

```diff
@@ -0,0 +1,785 @@
+        self.depth = 2
+
+
+class ResStemCifar(nn.Sequential):
+    """ResNet stem for CIFAR: 3x3, BN, ReLU."""
+
+    def __init__(
+        self,
+        width_in: int,
+        width_out: int,
+        bn_epsilon: float,
+        bn_momentum: float,
+        activation: nn.Module,
+    ) -> None:
+        super().__init__()
+        self.stem = nn.Sequential(
+            nn.Conv2d(width_in, width_out, 3, stride=1, padding=1, bias=False),
+            nn.BatchNorm2d(width_out, eps=bn_epsilon, momentum=bn_momentum),
+            activation,
+        )
+        self.depth = 2
+
+
+class ResStemIN(nn.Sequential):
+    """ResNet stem for ImageNet: 7x7, BN, ReLU, MaxPool."""
+
+    def __init__(
+        self,
+        width_in: int,
+        width_out: int,
+        bn_epsilon: float,
+        bn_momentum: float,
+        activation: nn.Module,
+    ) -> None:
+        super().__init__()
+        self.stem = nn.Sequential(
+            nn.Conv2d(width_in, width_out, 7, stride=2, padding=3, bias=False),
+            nn.BatchNorm2d(width_out, eps=bn_epsilon, momentum=bn_momentum),
+            activation,
+            nn.MaxPool2d(3, stride=2, padding=1),
+        )
+        self.depth = 3
+
+
+class SimpleStemIN(nn.Sequential):
+    """Simple stem for ImageNet: 3x3, BN, ReLU."""
+
+    def __init__(
+        self,
+        width_in: int,
+        width_out: int,
+        bn_epsilon: float,
+        bn_momentum: float,
+        activation: nn.Module,
+    ) -> None:
+        super().__init__()
+        self.stem = nn.Sequential(
+            nn.Conv2d(width_in, width_out, 3, stride=2, padding=1, bias=False),
+            nn.BatchNorm2d(width_out, eps=bn_epsilon, momentum=bn_momentum),
+            activation,
+        )
+        self.depth = 2
+
+
+class VanillaBlock(nn.Sequential):
+    """Vanilla block: [3x3 conv, BN, Relu] x2."""
+
+    def __init__(
+        self,
+        width_in: int,
+        width_out: int,
+        stride: int,
+        bn_epsilon: float,
+        bn_momentum: float,
+        activation: nn.Module,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__()
+        self.a = nn.Sequential(
+            nn.Conv2d(width_in, width_out, 3, stride=stride, padding=1, bias=False),
+            nn.BatchNorm2d(width_out, eps=bn_epsilon, momentum=bn_momentum),
+            activation,
+        )
+
+        self.b = nn.Sequential(
+            nn.Conv2d(width_out, width_out, 3, stride=1, padding=1, bias=False),
+            nn.BatchNorm2d(width_out, eps=bn_epsilon, momentum=bn_momentum),
+            activation,
+        )
+
+        self.depth = 2
+
+
+class ResBasicBlock(nn.Module):
+    """Residual basic block: x + F(x), F = basic transform."""
+
+    def __init__(
+        self,
+        width_in: int,
+        width_out: int,
+        stride: int,
+        bn_epsilon: float,
+        bn_momentum: float,
+        activation: nn.Module,
```

Comment:
Aha! Sorry I didn't realize, thanks for the clarification.

I think we avoid passing and reusing the same activation instance of an activation function because:
1. Some activation methods might potentially learn parameters (like [PReLU](https://pytorch.org/docs/stable/generated/torch.nn.PReLU.html#torch.nn.PReLU)) so at each place we might need different instances so that they don't share weights.
2. The Quantization API will have issues fusing the operators. We typically use the nn.* versions of them so that they can fuse them easier. 

As far as I see, you also make use of the `Callable[..., nn.Module]` [elsewhere](https://github.com/pytorch/vision/pull/4403/files/617b833a12c7f08d42b8b13d2037ddcf8075cb2c#diff-62577d15011d998701098e5e567a78bd117257deeeb2a23b72e550e1f6b14279R400), would you mind taking the same approach here?